account_bank_statement_import_camt/camt.py

```diff
@@ -118,27 +124,54 @@ def parse_transaction(self, ns, node):
             ns, node, './ns:BookgDt/ns:Dt', transaction, 'execution_date')
         self.add_value_from_node(
             ns, node, './ns:ValDt/ns:Dt', transaction, 'value_date')
+        amount = self.parse_amount(ns, node)
+        if amount != 0.0:
+            transaction['amount'] = amount
+        self.add_value_from_node(
+            ns, node, './ns:AddtlNtryInf', transaction, 'name')
+        self.add_value_from_node(
+            ns, node, [
+                './ns:NtryDtls/ns:RmtInf/ns:Strd/ns:CdtrRefInf/ns:Ref',
+                './ns:NtryDtls/ns:Btch/ns:PmtInfId',
+            ],
+            transaction, 'ref'
+        )
 
-        transaction['amount'] = self.parse_amount(ns, node)
-
-        details_node = node.xpath(
+        details_nodes = node.xpath(
             './ns:NtryDtls/ns:TxDtls', namespaces={'ns': ns})
-        if details_node:
-            self.parse_transaction_details(ns, details_node[0], transaction)
-        if not transaction.get('name'):
-            self.add_value_from_node(
-                ns, node, './ns:AddtlNtryInf', transaction, 'name')
-        if not transaction.get('name'):
-            transaction['name'] = '/'
-        if not transaction.get('ref'):
-            self.add_value_from_node(
-                ns, node, [
-                    './ns:NtryDtls/ns:Btch/ns:PmtInfId',
-                ],
-                transaction, 'ref'
-            )
-        transaction['data'] = etree.tostring(node)
-        return transaction
+        if len(details_nodes) == 0:
+            yield transaction
+            return
+        transaction_base = transaction
+        for i, dnode in enumerate(details_nodes):
+            transaction = transaction_base.copy()
+            self.parse_transaction_details(ns, dnode, transaction)
+            # transactions['data'] should be a synthetic xml snippet which
+            # contains only the TxDtls that's relevant.
+            data = copy(node)
+            for j, dnode in enumerate(data.xpath(
+                    './ns:NtryDtls/ns:TxDtls', namespaces={'ns': ns})):
+                if j != i:
+                    dnode.getparent().remove(dnode)
+            transaction['data'] = etree.tostring(data)
+            yield transaction
+
+    @staticmethod
+    def _deduplicate_names(transactions):
```

Comment:
I fail to see the general benefit of this change. It also might mess up automatic matching systems that rely on a specific reference format.

Maybe you can keep this in your own customizations? If you really want to share this, make it more general by implementing it at the Odoo bank statement level instead of the camt level.

Do you see my point, or am I missing yours?